db/document.go

```diff
@@ -92,14 +137,14 @@ func unmarshalDocument(docid string, data []byte) (*document, error) {
 	return doc, nil
 }
 
-func unmarshalDocumentWithXattr(docid string, data []byte, xattrData []byte, cas uint64) (doc *document, err error) {
+func unmarshalDocumentWithXattr(docid string, data []byte, xattrData []byte, cas uint64, unmarshalTo DocumentUnmarshalLevel) (doc *document, err error) {
```

Comment:
The param name and type name mismatch is a bit confusing.  Wondering if this would be clearer as:

```
unmarshalLevel DocumentUnmarshalLevel
```

or 

```
level DocumentUnmarshalLevel
```

When I see `unmarshalTo` I actually think of an unmarshal destination/target.  Eg, a stream or sink of some sort.

Reply:
Agreed - will make that change.